src/transformStateWithClones.js

```diff
@@ -8,6 +8,39 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const statesArray = [];
+  const cloneState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(cloneState, action.extraData);
+        break;
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete cloneState[key];
+        }
+
+        break;
+      }
+
+      case 'clear': {
+        for (const key of Object.keys(cloneState)) {
+          delete cloneState[key];
+        }
```

Comment:
```suggestion
        cloneState = {}
```

Reply:
cloneState in my case is constant, had to change it to let